### PR TITLE
NCBI datasets: Update macros.xml regarding taxon selection

### PR DESCRIPTION
**Pull request by AnnaSyme**
Suggest adding this text to the wrapper to clarify how to select a taxon. (If this is correct?)

FOR CONTRIBUTOR:
* [- ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [- ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/ncbi_datasets/macros.xml [hunk 1]
```diff
@@ -71,7 +71,7 @@
     </xml>
 
     <xml name="taxon_positional">
-        <param name="taxon_positional" type="text" label="Enter taxon" help="e.g. human, mouse, bos taurus, etc."/>
+        <param name="taxon_positional" type="text" label="Enter taxon" help="e.g. human, mouse, bos taurus, etc. or use Taxonomy ID from NCBI for greater accuracy, use species-level only not genera or larger taxa"/>
```
Comment (bernt-matthias) [inline tools/ncbi_datasets/macros.xml:74]: What do you mean by accuracy? Probably depends on the question.

I for instance used this tool only to download all the data for larger taxonomic groups. In this context the last part seems to make not so much sense.

The statement about taxids definitely makes sense.

**Comment by bgruening** [comment]
I'm not sure what the error means:

https://github.com/galaxyproject/tools-iuc/actions/runs/8107796267

Should I update the tool to the latest version? I fear that many more tests will fail ;)

File: tools/ncbi_datasets/macros.xml (hunk 1 above)
Comment (AnnaSyme) [reply tools/ncbi_datasets/macros.xml:74]: ah right, thanks @bernt-matthias - I thought it was not extracting data beyond the species level. I'm doing another test to see. Perhaps it's a different error, if I narrow it down I will post it as an issue for this tool.

**Comment by AnnaSyme** [comment]
> I'm not sure what the error means:
> 
> https://github.com/galaxyproject/tools-iuc/actions/runs/8107796267
> 
> Should I update the tool to the latest version? I fear that many more tests will fail ;)

Hi @bgruening sorry I can't see the error in that link, but I am getting an error that says ```New version of client (16.6.0) available at https://ftp.ncbi.nlm.nih.gov/pub/datasets/command-line/LATEST/linux-amd64/datasets
Error: No genes found that match selection

```

**Comment by bgruening** [comment]
@AnnaSyme can you look at the bottom of https://github.com/galaxyproject/tools-iuc/actions/runs/8114718425?pr=5803

**Comment by AnnaSyme** [comment]
> @AnnaSyme can you look at the bottom of https://github.com/galaxyproject/tools-iuc/actions/runs/8114718425?pr=5803

Sorry @bgruening I read it but I don't understand why it's failing

**Comment by bernt-matthias** [comment]
Will have a look.

**Review by bgruening — APPROVED** [review]

**Pull request merged**